app/controllers/provider_interface/sessions_controller.rb

```diff
@@ -16,7 +16,7 @@ def destroy
     end
 
     def sign_in_by_email
-      raise unless FeatureFlag.active?('dfe_sign_in_fallback')
+      render_404 unless FeatureFlag.active?('dfe_sign_in_fallback')
```

Comment:
I clicked on a magic link URL to QA, and it signed me in, even though the feature flag is off.

@tijmenb this renders a 404 and then proceeds to sign the user in anyway. This means that even if the feature flag is off, we accept the sign in attempt using the magic link. This seems incorrect; should it be:

```ruby
unless FeatureFlag.active?('dfe_sign_in_fallback')
  render_404
  return
end
```

Reply:
https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/2072